build_tools/docker/manage_images.py

```diff
@@ -0,0 +1,249 @@
+    sys.exit(exit_code)
+
+
+def get_repo_digest(image):
+  inspect_command = [
+      'docker',
+      'image',
+      'inspect',
+      f'{image}',
+      '-f',
+      '{{index .RepoDigests 0}}',
+  ]
+  inspect_process = subprocess.run(inspect_command,
+                                   text=True,
+                                   capture_output=True,
+                                   timeout=10)
+  if inspect_process.returncode != 0:
+    print(f'Computing the repository digest for {image} failed.'
+          ' Has it been pushed to GCR?')
+    print(f'Output from `{" ".join(inspect_command)}`:')
+    print(inspect_process.stdout, end='')
+    print(inspect_process.stderr, end='')
+    sys.exit(inspect_process.returncode)
+  _, repo_digest = inspect_process.stdout.strip().split('@')
+  return repo_digest
+
+
+def update_rbe_reference(digest, dry_run=False):
+  print('Updating WORKSPACE file for rbe-toolchain')
+  for line in fileinput.input(files=['WORKSPACE'], inplace=(not dry_run)):
+    if line.strip().startswith('digest ='):
+      print(re.sub('sha256:[a-zA-Z0-9]+', digest, line), end='')
+    else:
+      print(line, end='')
+
+
+def update_references(image_name, digest, dry_run=False):
+  print(f'Updating references to {image_name}')
+
+  # Just hardcode this oddity
+  if image_name.rsplit('/', maxsplit=1)[-1] == 'rbe-toolchain':
+    update_rbe_reference(digest, dry_run=dry_run)
+
+  grep_command = ['git', 'grep', '-l', f'{image_name}@sha256']
+  grep_process = subprocess.run(grep_command,
+                                capture_output=True,
+                                timeout=5,
+                                text=True)
+  if grep_process.returncode > 1:
+    print(f'{" ".join(grep_command)} '
+          f'failed with exit code {grep_process.returncode}')
+    sys.exit(grep_process.returncode)
+  if grep_process.returncode == 1:
+    print(f'Found no references to {image_name}')
+    return
+
+  files = grep_process.stdout.split()
+  print(f'Updating references in {len(files)} files: {files}')
+  for line in fileinput.input(files=files, inplace=(not dry_run)):
+    print(re.sub(f'{image_name}@sha256:[a-zA-Z0-9]+', f'{image_name}@{digest}', line),
+          end='')
+
+
+if __name__ == '__main__':
+  args = parse_arguments()
+
+  # Ensure the user has the correct authorization if they try to push to GCR.
+  if args.push:
+    if stream_command(['which', 'gcloud']) != 0:
+      print('gcloud not found.'
+            ' See https://cloud.google.com/sdk/install for installation.')
+      sys.exit(1)
+    check_stream_command(['gcloud', 'auth', 'configure-docker'],
+                         dry_run=args.dry_run)
+
+
+  images_to_process = get_ordered_images_to_process(args.images)
+  print(f'Also processing dependent images. Will process: {images_to_process}')
+
+  for image in images_to_process:
+    print(f'Processing image {image}')
+    image_name = os.path.join(IREE_GCR_URL, image)
```

Comment:
I'm pretty sure that this will be something like `'gcr.io/iree-oss/\cmake'` on windows... Could you chime in @ScottTodd?

I think this should work regardless:
```python3
import posixpath
image_name = posixpath.join(IREE_GCR_URL, image)
```

`urllib.parse.urljoin` is not ideal because of its behavior when a trailing `/` isn't provided: `urllib.parse.urljoin('gcr.io/iree-oss', image) == f'gcr.io/{image}'`

Reply:
I think this is the same as what it was before. But this isn't really even a filepath, so maybe we can just manually insert the slash :-D

Reply:
Used posixpath